projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -0,0 +1,1390 @@
+	 *
+	 * @global $wp_additional_image_sizes
+	 * @uses get_option
+	 * @return array
+	 */
+	protected static function image_sizes() {
+		if ( null === self::$image_sizes ) {
+			global $_wp_additional_image_sizes;
+
+			// Populate an array matching the data structure of $_wp_additional_image_sizes so we have a consistent structure for image sizes.
+			$images = array(
+				'thumb'        => array(
+					'width'  => (int) get_option( 'thumbnail_size_w' ),
+					'height' => (int) get_option( 'thumbnail_size_h' ),
+					'crop'   => (bool) get_option( 'thumbnail_crop' ),
+				),
+				'medium'       => array(
+					'width'  => (int) get_option( 'medium_size_w' ),
+					'height' => (int) get_option( 'medium_size_h' ),
+					'crop'   => false,
+				),
+				'medium_large' => array(
+					'width'  => (int) get_option( 'medium_large_size_w' ),
+					'height' => (int) get_option( 'medium_large_size_h' ),
+					'crop'   => false,
+				),
+				'large'        => array(
+					'width'  => (int) get_option( 'large_size_w' ),
+					'height' => (int) get_option( 'large_size_h' ),
+					'crop'   => false,
+				),
+				'full'         => array(
+					'width'  => null,
+					'height' => null,
+					'crop'   => false,
+				),
+			);
+
+			// Compatibility mapping as found in wp-includes/media.php.
+			$images['thumbnail'] = $images['thumb'];
+
+			// Update class variable, merging in $_wp_additional_image_sizes if any are set.
+			if ( is_array( $_wp_additional_image_sizes ) && ! empty( $_wp_additional_image_sizes ) ) {
+				self::$image_sizes = array_merge( $images, $_wp_additional_image_sizes );
+			} else {
+				self::$image_sizes = $images;
+			}
+		}
+
+		return is_array( self::$image_sizes ) ? self::$image_sizes : array();
+	}
+
+	/**
+	 * Pass og:image URLs through Photon
+	 *
+	 * @param array $tags Open graph tags.
+	 * @param array $parameters Image parameters.
+	 * @uses jetpack_photon_url
+	 * @return array Open graph tags.
+	 */
+	public function filter_open_graph_tags( $tags, $parameters ) {
+		if ( empty( $tags['og:image'] ) ) {
+			return $tags;
+		}
+
+		$photon_args = array(
+			'fit' => sprintf( '%d,%d', 2 * $parameters['image_width'], 2 * $parameters['image_height'] ),
+		);
+
+		if ( is_array( $tags['og:image'] ) ) {
+			$images = array();
+			foreach ( $tags['og:image'] as $image ) {
+				$images[] = jetpack_image_cdn_url( $image, $photon_args );
+			}
+			$tags['og:image'] = $images;
+		} else {
+			$tags['og:image'] = jetpack_image_cdn_url( $tags['og:image'], $photon_args );
+		}
+
+		return $tags;
+	}
+
+	/**
+	 * Enqueue Photon helper script
+	 *
+	 * @uses wp_enqueue_script, plugins_url
+	 * @action wp_enqueue_script
+	 * @return null
+	 */
+	public function action_wp_enqueue_scripts() {
+		if ( self::is_amp_endpoint() ) {
+			return;
+		}
+		wp_enqueue_script(
+			'jetpack-photon',
+			Assets::get_file_url_for_environment(
+				'js/photon.min.js',
```

Comment:
I note that you don't have any build steps actually creating the minified file. When you do add the build steps, I'd recommend not putting them at this location as it seems to work out cleaner when there's a `build/` or `dist/` directory for built files instead. The lazy-images package might work as an example.

Also, when you do add that build step, it'll probably be worth building in the more modern manner so you can use `Assets::register_script()` instead of this old way.

(for that matter, someday we should go through Jetpack's modules and do the same now that wpcom isn't being weird about them anymore, but that's way off-topic for this PR)